airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -651,6 +657,27 @@ public void writeCatalog(final AirbyteCatalog catalog,
         actorCatalogFetchEvent);
   }
 
+  private SelectConditionStep selectCountWorkspaceConnections(final DSLContext ctx, final UUID workspaceId) {
+    return ctx.selectCount()
+        .from(ACTOR).join(WORKSPACE)
+        .on(ACTOR.WORKSPACE_ID.equal(WORKSPACE.ID))
+        .where(WORKSPACE.ID.equal(workspaceId));
+  }
+
+  public int countConnectionsForWorkspace(final UUID workspaceId) throws IOException {
+    return database.query(ctx -> selectCountWorkspaceConnections(ctx, workspaceId)).fetchOne().into(int.class);
```

Comment:
As far as I understand, this would count the total amount of sources and destinations, but not the amount of connections (which should be stored in the `connection` table)?

Reply:
My bad, it seems I misunderstood the needs expressed in that review: https://github.com/airbytehq/airbyte/pull/10071/commits/89720b1b9ec5231c5020c9a60fcc18ae08143fa1. 
So I guess you need a count of sources, destinations and connections.

Reply:
Please take another look